pkg/common/cryptoutil/keys.go

```diff
@@ -48,3 +52,23 @@ func KeyMatches(privateKey crypto.PrivateKey, publicKey crypto.PublicKey) (bool,
 		return false, fmt.Errorf("unsupported private key type %T", privateKey)
 	}
 }
+
+func GenerateSigner(keyType keymanagerv0.KeyType) (signer crypto.Signer, err error) {
+	switch keyType {
+	case keymanagerv0.KeyType_EC_P256:
+		return ecdsa.GenerateKey(elliptic.P256(), rand.Reader)
+	case keymanagerv0.KeyType_EC_P384:
+		return ecdsa.GenerateKey(elliptic.P384(), rand.Reader)
+	case keymanagerv0.KeyType_RSA_2048:
+		return rsa.GenerateKey(rand.Reader, 2048)
+	case keymanagerv0.KeyType_RSA_4096:
+		return rsa.GenerateKey(rand.Reader, 4096)
+	case keymanagerv0.KeyType_UNSPECIFIED_KEY_TYPE:
+		// Default to ECDSA with P256 curve
+		return ecdsa.GenerateKey(elliptic.P256(), rand.Reader)
+	case keymanagerv0.KeyType_RSA_1024:
+		// We do not allow configuring this key type
+		return nil, fmt.Errorf("unsupported key type %q", keyType)
```

Comment:
These two details (default to EC256 and no support for RSA1024) seem inappropriate from a "generic helper to generate a signer from a key type" perspective.

The default is applied down in the ca manager code, but perhaps now that we have a dependency on the key type a little higher in the stack, we should apply the default earlier? I'm thinking in the cli code for the run command here: https://github.com/spiffe/spire/blob/2636b93c4052e85e648458d4ab17acf191a87f52/cmd/spire-server/cli/run/run.go#L464

Reply:
Initially I did add this but one of the linters complains about creating rsa keys with 1024 bits so I changed it to this error. Seemed better than falling back to the default error of "unknown".

Sure, we can set a sane default earlier.